### DISCUSSION TITLE
Organizations

### DISCUSSION THREAD

**Discussion by igboyes** [virtool/.github#2] [Ideas]
Organizations will provided a new organizational level for users and data.

# Fields
| Field | Description |
|:------|:-------------|
| id | the unique ID for the organization |
| name | display name for the organization |
| description | longer description for the organization |
| created_at | when the organization was created |
| updated_at | when the organization was updated |
| created_by | the user that created the organization |

## Roll-out
When organizations are released:

### Administrators
* Administrators get all app-level permissions and the `administrator` flag is deprecated.
* Administrators can create organizations or allow all users to create organizations.
* Administrators can create users or allow other users to create users

### Groups
* Will still be called groups in code. We'll call subsets of organization members 'teams'.
* Groups will be ad hoc groups of users that can share data. They will be used for sharing data between organizations and users.
* Groups will **not** be convertable to organizations. This is too complex for references.

### Users
* Users are unattached to organizations on release.
* They own all data they created. Other users and groups with access to a resource will see it as a share.

### References
* They will be owned by the creating user.
* Existing user and group access will be presented as data shares.
* They will be transferrable to organizations.

### Samples
* Group ownership will be converted to group shares that mirror those used for references.
* Wil be transferrable to an organization. Shares will be preserved, but we should encourage users to clean these up. We could impose and expiry date and require opt-in to preserve the shares.

### Subtractions
Existing subtractions will be:
* Publicly visible.
* Only editable by the owner user.
* Transferrable to an organization.

### Uploads
* Will be owned by users.
* Transferrable to organizations.

## Users
- Users can be members of multiple organizations
- Users can have permissions in an organization
- Users can exist outside organizations and can generate an organization from their existing account

## Teams
- Teams have a one-to-one relationship with organizations
- Teams can have permissions that are inherited by their member users

## Logo
* Allow in-app logo upload and cropping
* The logo is used in the UI for quick identification of an organization.

**Comment by MikeERott** [comment]
would be good to have a way to share selected sample data across organizations. Or, for example, samples are tagged to a project that is across multiple organizations.

**Reply by igboyes** [reply]
That will probably have to be a separate discussion.
